pkg/webhook/controlplane/ensurer.go

```diff
@@ -79,70 +100,136 @@ var (
+	if makeAuditForwarder {
+		c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, auditPolicyVolumeMount)
+		c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, auditLogVolumeMount)
 	}
 }
 
-func ensureVolumes(ps *corev1.PodSpec, controllerConfig config.ControllerConfiguration) {
+func ensureVolumes(ps *corev1.PodSpec, makeAuditForwarder bool, controllerConfig config.ControllerConfiguration) {
 	if controllerConfig.Auth.Enabled {
 		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, authnWebhookConfigVolume)
 		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, authnWebhookCertVolume)
 	}
-	if controllerConfig.SplunkAudit.Enabled {
-		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, splunkAuditWebhookConfigVolume)
-		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, splunkAuditWebhookCertVolume)
+	if makeAuditForwarder {
+		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, auditPolicyVolume)
+		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, auditLogVolume)
+		ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, audittailerClientSecretVolume)
 	}
 }
 
-func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, controllerConfig config.ControllerConfiguration) {
+func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, makeAuditForwarder bool, controllerConfig config.ControllerConfiguration) {
 	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "external")
 
 	if controllerConfig.Auth.Enabled {
 		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--authentication-token-webhook-config-file=", "/etc/webhook/config/authn-webhook-config.json")
 	}
 
-	if controllerConfig.SplunkAudit.Enabled {
-		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-policy-file=", "/etc/splunkauditwebhook/config/audit-policy.yaml")
-		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-webhook-config-file=", "/etc/splunkauditwebhook/config/splunk-audit-webhook-kubeconfig")
-		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-webhook-batch-max-size=", "4") // To not exceed the splunk hec endpoint maximum event size. Maybe make it configurable? CHECKME
+	if makeAuditForwarder {
+		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-policy-file=", "/etc/kubernetes/audit-override/audit-policy.yaml")
+		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-log-path=", "/auditlog/audit.log")
+	}
+
+}
+
+func ensureAuditForwarder(ps *corev1.PodSpec, controllerConfig config.ControllerConfiguration) {
+	auditForwarderImage, err := imagevector.ImageVector().FindImage("auditforwarder")
+	if err != nil {
+		logger.Error(err, "Could not find auditforwarder image in imagevector")
```

Comment:
Would be a little better if an error was returned here, I guess.

Reply:
I changed it; is this what you had in mind?

Reply:
Yep, it's good.